### PR TITLE
docs: added typescript installation guide in INSTALLATION.md

### LINKED ISSUE #1243: The `INSTALLATION.md` file doesn't tell you to install typescript
**Is your feature request related to a problem? Please describe.**
1. The `INSTALLATION.md` file doesn't tell you how to install typescript
1. This is an obvious omission and must be fixed

**Describe the solution you'd like**
1. Update the `INSTALLATION.md` file to explain how to install typescript in the correct order after installing `npm`
2. Make this a separate section in the procedure
3. Ensure that the table of contents is correctly updated

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by MahendraDani**
**What kind of change does this PR introduce?**
docs

**Issue Number:**

Fixes #1243 

**Did you add tests for your changes?**
No

**Snapshots/Videos:**
NA

**If relevant, did you update the documentation?**

**Summary**
1. Added typescript installation section in `INSTALLATION.md`
2. Updated `table of content` in the file.

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by palisadoes** [comment]
Please look at the more comprehensive documentation for Talawa-API that explains how to install node and make the appropriate changes at the appropriate locations
- https://github.com/PalisadoesFoundation/talawa-api/blob/develop/INSTALLATION.md#install-nodejs

**Comment by MahendraDani** [comment]
> Please look at the more comprehensive documentation for Talawa-API that explains how to install node and make the appropriate changes at the appropriate locations
> 
> * https://github.com/PalisadoesFoundation/talawa-api/blob/develop/INSTALLATION.md#install-nodejs

@palisadoes I have following two doubts:
1. Should I update the current *Nodejs setup guide* as given in https://github.com/PalisadoesFoundation/talawa-api/blob/develop/INSTALLATION.md#install-nodejs ?
2. Is the setup guide for typescript written by me appropriate or it requires changes?

Thanks

**Comment by palisadoes** [comment]
Use what we have in the API guide as information in the Admin guide

**Comment by MahendraDani** [comment]
I will do the necessary changes and update the PR by tomorrow. 
Sorry for the delay

**Comment by MahendraDani** [comment]
@palisadoes I have updated the docs. I have referred to installation guide as provided in https://github.com/PalisadoesFoundation/talawa-api/blob/main/INSTALLATION.md.

If there are any changes please let me know.
Thanks

File: INSTALLATION.md [hunk 1]
```diff
@@ -62,7 +65,12 @@ cd talawa-admin
 
 ## Setting up NPM
 
-If you've followed the previous steps you should have already set up node.js on your system. [Click here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) for the official setup guide for npm.
+Best way to install and manage `node.js` is making use of node version managers. Two most popular node version managers right now are [fnm](https://github.com/Schniz/fnm) and [nvm](https://github.com/nvm-sh/nvm). We'd recommend `fnm` because it's written in `rust` and is much faster than `nvm`. Install whichever one you want and follow their guide to set up `node.js` on your system.
+
+## Setting up Typescript
+
+As `talawa-admin` and `talawa-api` repositories are written using [Typescript](https://www.typescriptlang.org/), you will need to install typescript on your machine.
```
Comment (palisadoes) [inline INSTALLATION.md:72]: 1. `fnm` and `nvm` install node and the various modules locally and we recommend using them.
2. Add the command to install typescript. Use the API INSTALLATION.md as a guide

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: INSTALLATION.md (hunk 1 above)
Comment (MahendraDani) [reply INSTALLATION.md:72]: @palisadoes 
`fnm` and `nvm` are version managers for `nodejs` which help to run multiple versions of `node` on a single machine whereas to install typescript we will have to use `npm` as given in the official typescript [docs](https://www.typescriptlang.org/download).
Refer : https://medium.com/@fs6158857/what-difference-between-npm-and-nvm-fa43e7631543#:~:text=In%20summary%2C%20NPM%20is%20a,play%20crucial%20roles%20in%20Node.
---
I will make the following changes in `Setting up typescript` section:
As `talawa-admin` and `talawa-api` repositories are written using `typescript`, you will need to install `typescript` on your machine. We recommend to install `typescript` globally using the following command
```
npm install typescript
```

You can also refer to the official [installation guide](https://www.typescriptlang.org/download)

Sir is this correct?

Comment (palisadoes) [reply INSTALLATION.md:72]: OK

**Comment by MahendraDani** [comment]
@palisadoes I have made the necessary changes

**Review by palisadoes — DISMISSED** [review]

**Comment by palisadoes** [comment]
Please fix the conflicting files

**Comment by MahendraDani** [comment]
@palisadoes I will close this PR and open a new full request to avoid merge conflicts.

**Comment by MahendraDani** [comment]
@palisadoes I have resolved merge conflicts and updated this branch with `develop` branch.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**